src/Microsoft.Performance.SDK/Processing/DirectoryDataSourceAttribute.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.Performance.SDK.Processing
+{
+    /// <summary>
+    ///     Attribute to mark a Custom Data Source implementation as processing
+    ///     directories.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = false)]
+    public sealed class DirectoryDataSourceAttribute
+        : DataSourceAttribute,
+          IEquatable<DirectoryDataSourceAttribute>
+    {
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="DirectoryDataSourceAttribute"/>
+        ///     class.
+        /// </summary>
+        public DirectoryDataSourceAttribute()
```

Comment:
I would argue we shouldn't have this ctor. We should enforce plugins specifying what the heck their supported directories represent... at least with files the extension can give some insight to users 😄